src/main/java/org/commcare/suite/model/Text.java

```diff
@@ -302,6 +334,46 @@ public boolean rawArgs() {
         }
     }
 
+    private String[] generateOrderedParameterListForLocalization(Hashtable<String, Text> arguments,
+                                                                 EvaluationContext context) {
+        if(arguments == null) {
+            return new String[0];
+        }
+
+        List<String> keys = getOrderedKeys(arguments);
+
+        if(keys.size() == 0) {
+            return new String[0];
+        }
+
+        String[] parameters = new String[keys.size()];
+        for(int i = 0; i < keys.size(); ++i) {
+            parameters[i] = arguments.get(keys.get(i)).evaluate(context);
+        }
+
+        return parameters;
+    }
+
+    private List<String> getOrderedKeys(Hashtable<String, Text> arguments) {
+        List<String> keys = new ArrayList<>();
+
+        for(String key : arguments.keySet()) {
+            if(key.equals("id")) {
+                continue;
+            }
+            keys.add(key);
+        }
+
+        //We presume all of the keys here are numbers and that they are in order (implicit in
```

Comment:
Looks like this comment was meant to be for L373 below ?

Reply:
not quite, I'm explaining why we error out if there are more than 10 keys, but not clearly enough. Will update.